HunterPie.Integrations/Datasources/MonsterHunterRise/Entity/Player/MHRPlayer.cs

```diff
@@ -390,12 +392,57 @@ private void GetConsumableAbnormalities()
 
             MHRConsumableStructure abnormality = new();
 
-            if (schema.IsInfinite)
-                abnormality.Timer = abnormSubId == schema.WithValue ? AbnormalityData.TIMER_MULTIPLIER : 0;
-            else if (abnormSubId == schema.WithValue)
-                abnormality = Process.Memory.Read<MHRConsumableStructure>(consumableBuffs + schema.Offset);
+            switch (schema.Id)
+            {
+                case "ABN_AGITATOR":
+                    abnormality.Timer = _commonCondition.HasFlag(CommonConditions.Agitator) ? 1 : 0;
+                    break;
+                case "ABN_DEFIANCE":
+                    abnormality.Timer = _commonCondition.HasFlag(CommonConditions.Defiance) ? 1 : 0;
+                    break;
+                case "ABN_HEROICS":
+                    abnormality.Timer = _commonCondition.HasFlag(CommonConditions.Heroics) ? 1 : 0;
+                    break;
+                case "ABN_PEAK_PERFORMANCE":
+                    abnormality.Timer = _commonCondition.HasFlag(CommonConditions.PeakPerformance) ? 1 : 0;
+                    break;
+                case "ABN_DANGO_DEFENDER":
+                    abnormality.Timer = _commonCondition.HasFlag(CommonConditions.DangoDefender) ? 1 : 0;
+                    break;
+                case "ABN_DRAGONHEART":
+                    abnormality.Timer = _commonCondition.HasFlag(CommonConditions.DragonHeart) ? 1 : 0;
+                    break;
+                case "ABN_RUBY_WIREBUG":
+                    if (_commonCondition.HasFlag(CommonConditions.MarionetteTypeRuby))
```

Comment:
Like I mentioned in a previous PR you opened, the abnormality scanners should be generic and agnostic to what abnormalities they're currently scanning. Filling the code with `switch/case` for handling each one of the abnormalities is the opposite of generic.
Specifications of each abnormality should be kept in the `AbnormalityData.xml`.

Reply:
Scanner edited.